addons/sale/models/product_template.py

```diff
@@ -242,28 +243,26 @@ def _get_combination_info(self, combination=False, product_id=False, add_qty=1,
             product_template = product_template.with_context(current_attributes_price_extra=current_attributes_price_extra)
             price_extra = sum(current_attributes_price_extra)
             list_price = product_template.price_compute('list_price')[product_template.id]
-            if pricelist:
-                price = pricelist._get_product_price(product_template, quantity)
-            else:
-                price = list_price
+            currency = pricelist.currency_id or currency or product_template.currency_id
+            price = pricelist._get_product_price(product_template, quantity, currency)
             display_image = bool(product_template.image_128)
-
             combination_name = combination._get_combination_name()
             if combination_name:
                 display_name = "%s (%s)" % (display_name, combination_name)
+        # TODO edm: Hiya Victor, cleaner code, slightly less efficient/direct, your opinion?
```

Comment:
My opinion is that a future task should refactor the price/discount/discounted_price computation to factorize this point as well :D.

Changes lgtm, but as said in other comments, currency spec could probably be simplified (and extracted from the `if product ... else` because we never want to fallback on the product currency anyway, it should be either the pricelist currency or the company currency)